core/filter_api.php

```diff
@@ -2225,21 +2225,25 @@ function filter_get_bug_rows_query_clauses( array $p_filter, $p_project_id = nul
 
 /**
  *  Cache the filter results with bugnote stats for later use
+ *  also fills bug attachment count cache
  * @param array $p_rows             Results of the filter query.
  * @param array $p_id_array_lastmod Array of bug ids.
  * @return array
  */
 function filter_cache_result( array $p_rows, array $p_id_array_lastmod ) {
 	$t_stats = bug_get_bugnote_stats_array( $p_id_array_lastmod );
 	$t_rows = array();
+	$t_bug_ids = array();
 	foreach( $p_rows as $t_row ) {
-		$b = $t_stats[$t_row['id']];
+		$t_bug_ids[] = (int)$t_row['id'];
 		if( array_key_exists( $t_row['id'], $t_stats ) ) {
 			$t_rows[] = bug_row_to_object( bug_cache_database_result( $t_row, $t_stats[$t_row['id']] ) );
 		} else {
 			$t_rows[] = bug_row_to_object( bug_cache_database_result( $t_row ) );
 		}
 	}
+	#cache the attachment count for bug list
+	file_bug_attachment_count_cache( $t_bug_ids );
```

Comment:
I think he means that in bugnote_clear_bug_cache(), all entries in $g_cache_bugnote for the corresponding bug_id are already unset (see [line 783](https://github.com/mantisbt/mantisbt/pull/941/files#diff-af4b0457065b17f59cdc20eec8005bc4R783))